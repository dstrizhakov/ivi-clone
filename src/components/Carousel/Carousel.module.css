.carousel {
  overflow: hidden;
  padding: 10px;
}
.carousel .inner {
  white-space: nowrap;
  transition: transform 1s;
  flex-direction: row;
  display: flex;
}
.carousel .arrows {
  display: flex;
  justify-content: space-between;
}
.carousel .arrow {
  position: relative;
  top: -250px;
  cursor: pointer;
  font-weight: 400;
  font-style: normal;
  font-size: 32px;
  line-height: 32px;
  speak: none;
  border: 0;
  background-color: transparent;
  color: white;
}
.carousel .arrow:disabled {
  color: transparent;
}

/*# sourceMappingURL=Carousel.module.css.map */
