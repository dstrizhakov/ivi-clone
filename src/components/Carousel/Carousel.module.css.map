{"version":3,"sourceRoot":"","sources":["Carousel.module.scss"],"names":[],"mappings":"AAAA;EACE;EACA;;AACA;EACE;EACA;EACA;EACA;;AAEF;EACE;EACA;;AAEF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACE","file":"Carousel.module.css"}