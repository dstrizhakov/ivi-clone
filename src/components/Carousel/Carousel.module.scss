@import '@/styles/variables.scss';

.carousel {
  -webkit-tap-highlight-color: transparent;
  margin-bottom: 20px;
  .title {
    max-width: 30%;
    display: flex;
    align-items: center;
    svg {
      min-width: 25px;
    }
  }
}
