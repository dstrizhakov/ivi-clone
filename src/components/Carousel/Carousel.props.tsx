import { ReactNode } from 'react';
import 'i18next';

export interface CarouselProps {
  title?: string;
  route?: string;
  showAll?: boolean;
  children: ReactNode;
  settings?: object;
}
