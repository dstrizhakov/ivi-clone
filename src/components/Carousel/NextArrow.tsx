import styles from '@/components/Carousel/Carousel.module.scss';
import { Button } from '@/components/Button/Button';
import { MdArrowForwardIos } from 'react-icons/md';
import React, { FC } from 'react';
import { BtnA, BtnS } from '@/components/Button/Button.props';

export type NextArrowProps = {
  variant?: 'promo';
};
const NextArrow: FC<NextArrowProps> = ({ variant, ...props }): JSX.Element => {
  return (
    <div onClick={props.onClick} className={styles.arrow_container}>
      <div className={`${styles.arrow} ${variant === 'promo' ? styles.nextPromo : styles.next}`}>
        <Button
          size={BtnS.L}
          appearance={BtnA.transparent}
          disabled={props.className.split(' ')[2] === 'slick-disabled'}
        >
          <MdArrowForwardIos size={30} />
        </Button>
      </div>
      {variant == 'promo' && <div className={styles.right_banner} />}
    </div>
  );
};

export default NextArrow;
