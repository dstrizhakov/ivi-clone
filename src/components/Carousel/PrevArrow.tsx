import styles from '@/components/Carousel/Carousel.module.scss';
import { Button } from '@/components/Button/Button';
import { MdArrowBackIosNew } from 'react-icons/md';
import { FC } from 'react';
import { BtnA, BtnS } from '@/components/Button/Button.props';

export type PrevArrowProps = {
  variant?: 'promo';
};

const PrevArrow: FC<PrevArrowProps> = ({ variant, ...props }): JSX.Element => {
  return (
    <div
      className={`${styles.arrow} ${variant === 'promo' ? styles.prevPromo : styles.prev}`}
      onClick={props.onClick}
    >
      <Button
        size={BtnS.L}
        appearance={BtnA.transparent}
        disabled={props.className.split(' ')[2] === 'slick-disabled'}
      >
        <MdArrowBackIosNew size={30} />
      </Button>
    </div>
  );
};

export default PrevArrow;
