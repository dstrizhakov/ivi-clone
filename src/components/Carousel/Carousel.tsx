import React, { FC } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import styles from './Carousel.module.scss';
import { MdArrowForwardIos } from 'react-icons/md';
import { CarouselProps } from '@/components/Carousel/Carousel.props';
import Link from 'next/link';
import { Htag } from '@/components/Htag/Htag';
import ShowAll from '@/components/Card/ShowAll';
import NextArrow from '@/components/Carousel/Arrows/NextArrow';
import PrevArrow from '@/components/Carousel/Arrows/PrevArrow';

const Carousel: FC<CarouselProps> = ({
  title,
  route,
  showAll = false,
  children,
  settings,
}): JSX.Element => {
  if (!settings)
    settings = {
      dots: false,
      infinite: false,
      speed: 500,
      slidesToShow: 7,
      draggable: true,
      lazyLoad: 'progressive',
      slidesToScroll: 6,
      nextArrow: <NextArrow />,
      prevArrow: <PrevArrow />,
      responsive: [
        {
          breakpoint: 1160,
          settings: {
            slidesToShow: 6,
            slidesToScroll: 5,
          },
        },
        {
          breakpoint: 1095,
          settings: {
            slidesToShow: 5,
            slidesToScroll: 4,
          },
        },
        {
          breakpoint: 920,
          settings: {
            slidesToShow: 4,
            slidesToScroll: 3,
          },
        },
        {
          breakpoint: 750,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 2,
          },
        },
        {
          breakpoint: 600,
          settings: {
            slidesToShow: 4,
            slidesToScroll: 1,
          },
        },
        {
          breakpoint: 510,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 2,
          },
        },
        {
          breakpoint: 390,
          settings: {
            slidesToShow: 2,
            slidesToScroll: 1,
          },
        },
        {
          breakpoint: 320,
          settings: {
            slidesToShow: 1,
            slidesToScroll: 1,
            arrows: false,
          },
        },
      ],
    };

  return (
    <div className={styles.carousel}>
      {title && (
        <div>
          <Link href={route || ''} className={styles.title}>
            <div title={title}>
              <Htag tag={'h4'}>{title}</Htag>
            </div>
            <MdArrowForwardIos />
          </Link>
          <div />
        </div>
      )}
      <Slider {...settings}>
        {children}
        {showAll && (
          <Link href={route || ''}>
            <ShowAll />
          </Link>
        )}
      </Slider>
    </div>
  );
};

export default Carousel;
