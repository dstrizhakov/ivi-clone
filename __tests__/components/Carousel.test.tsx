import { render } from '@testing-library/react';
import React from 'react';
import Carousel from '../../src/components/Carousel/Carousel';

const testArray = [1, 2, 3, 4, 5];

describe('carousel', () => {
  test('number array', () => {
    const carousel = render(
      <Carousel title={'title'}>
        <>{testArray}</>
      </Carousel>
    );
    expect(carousel).toMatchSnapshot();
  });
  test('number array show all', () => {
    const carousel = render(
      <Carousel title={'title'} showAll>
        <>{testArray}</>
      </Carousel>
    );
    expect(carousel).toMatchSnapshot();
  });
  test('number array show all', () => {
    const carousel = render(
      <Carousel title={'title'} showAll>
        <>{testArray}</>
      </Carousel>
    );
    expect(carousel).toMatchSnapshot();
  });
});
