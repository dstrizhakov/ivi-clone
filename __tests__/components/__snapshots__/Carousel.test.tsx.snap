// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel number array 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div>
          <a
            class="title"
            href=""
          >
            <div
              title="title"
            >
              <h4
                class="h h4"
              >
                title
              </h4>
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </a>
          <div />
        </div>
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  1
                  2
                  3
                  4
                  5
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div>
        <a
          class="title"
          href=""
        >
          <div
            title="title"
          >
            <h4
              class="h h4"
            >
              title
            </h4>
          </div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </a>
        <div />
      </div>
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                1
                2
                3
                4
                5
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`carousel number array show all 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div>
          <a
            class="title"
            href=""
          >
            <div
              title="title"
            >
              <h4
                class="h h4"
              >
                title
              </h4>
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </a>
          <div />
        </div>
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  1
                  2
                  3
                  4
                  5
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <a
                    href=""
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="imageSection show"
                      >
                        <p
                          class="p M"
                        >
                          buttons.show-all
                        </p>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div>
        <a
          class="title"
          href=""
        >
          <div
            title="title"
          >
            <h4
              class="h h4"
            >
              title
            </h4>
          </div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </a>
        <div />
      </div>
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                1
                2
                3
                4
                5
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <a
                  href=""
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection show"
                    >
                      <p
                        class="p M"
                      >
                        buttons.show-all
                      </p>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`carousel number array show all 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div>
          <a
            class="title"
            href=""
          >
            <div
              title="title"
            >
              <h4
                class="h h4"
              >
                title
              </h4>
            </div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </a>
          <div />
        </div>
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  1
                  2
                  3
                  4
                  5
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <a
                    href=""
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="imageSection show"
                      >
                        <p
                          class="p M"
                        >
                          buttons.show-all
                        </p>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div>
        <a
          class="title"
          href=""
        >
          <div
            title="title"
          >
            <h4
              class="h h4"
            >
              title
            </h4>
          </div>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </a>
        <div />
      </div>
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                1
                2
                3
                4
                5
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <a
                  href=""
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <div
                    class="card"
                  >
                    <div
                      class="imageSection show"
                    >
                      <p
                        class="p M"
                      >
                        buttons.show-all
                      </p>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
